front/src/modules/activities/tasks/hooks/useTasks.ts

```diff
@@ -1,47 +1,19 @@
-import { useEffect } from 'react';
 import { DateTime } from 'luxon';
-import { useRecoilState } from 'recoil';
 
 import { TasksRecoilScopeContext } from '@/activities/states/recoil-scope-contexts/TasksRecoilScopeContext';
-import { useInitializeTasksFilters } from '@/activities/tasks/hooks/useInitializeTasksFilters';
 import { ActivityTargetableEntity } from '@/activities/types/ActivityTargetableEntity';
-import { currentUserState } from '@/auth/states/currentUserState';
 import { useRecoilScopedState } from '@/ui/utilities/recoil-scope/hooks/useRecoilScopedState';
 import { filtersScopedState } from '@/ui/view-bar/states/filtersScopedState';
-import { FilterOperand } from '@/ui/view-bar/types/FilterOperand';
 import { turnFilterIntoWhereClause } from '@/ui/view-bar/utils/turnFilterIntoWhereClause';
 import { ActivityType, useGetActivitiesQuery } from '~/generated/graphql';
-import { tasksFilters } from '~/pages/tasks/tasks-filters';
 import { parseDate } from '~/utils/date-utils';
 
 export function useTasks(entity?: ActivityTargetableEntity) {
-  useInitializeTasksFilters({
-    availableFilters: tasksFilters,
-  });
-
-  const [filters, setFilters] = useRecoilScopedState(
+  const [filters] = useRecoilScopedState(
     filtersScopedState,
     TasksRecoilScopeContext,
   );
 
-  // If there is no filter, we set the default filter to the current user
-  const [currentUser] = useRecoilState(currentUserState);
```

Comment:
extracting this out of useTasks as this will trigger re-render. I'm moving it to a TasksEffect component. only this TasksEffect component will be re-rendered.

Reply:
Makes sense